sklearn/tree/_criterion.pyx

```diff
@@ -309,24 +310,18 @@ cdef class ClassificationCriterion(Criterion):
         self.end = end
         self.n_node_samples = end - start
         self.weighted_n_samples = weighted_n_samples
-        cdef double weighted_n_node_samples = 0.0
+        self.weighted_n_node_samples = 0.0
 
-        cdef SIZE_t n_outputs = self.n_outputs
-        cdef SIZE_t* n_classes = self.n_classes
-        cdef SIZE_t label_count_stride = self.label_count_stride
-        cdef double* label_count_total = self.label_count_total
+        cdef double* node_sum = self.node_sum
 
-        cdef SIZE_t i = 0
-        cdef SIZE_t p = 0
-        cdef SIZE_t k = 0
-        cdef SIZE_t c = 0
+        cdef SIZE_t i, p, k, c
```

Comment:
For me the rule is that if they are used as iterators or indexes then they can go on the same line, since they are just helper variables. Having the variables on different lines makes the functions look, to me, more complicated than they actually are. However, I don't feel too strongly about it so I can change it back if you disagree.